lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -953,6 +958,28 @@ ChangeResult TypeAnalyzer::visitAtenMaxPool2dOp(
   return getLatticeElement(op->getResult(0)).join(knowledge);
 }
 
+ChangeResult TypeAnalyzer::visitAtenConstantPadNdOp(
+    AtenConstantPadNdOp op,
+    ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto knowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  knowledge.hasSizes = true;
+  auto &ifm = operands[0]->getValue();
+  SmallVector<int64_t, 4> padInts;
+  if (ifm.hasSizes && matchPattern(op.pad(), m_TorchConstantIntList(padInts))) {
+    knowledge.sizes = ifm.sizes;
+    uint64_t padRank = padInts.size() / 2;
+    uint64_t padOffset = knowledge.sizes.size() - padRank;
+    // op.pad() is higest dim first ordered pairs of low,high
+    for (uint64_t i = padRank, r = padOffset; i > 0; --i, ++r) {
+      knowledge.sizes[r] += padInts[i * 2 - 2] + padInts[i * 2 - 1];
+    }
+  } else
+    knowledge.sizes.resize(4, kUnknownSize);
```

Comment:
we don't need the else branch. The `knowledge` is initialized with unknown size. If you would like to be more accurate. You could arrange the branches like
```
if (input.hasSizes) {
   knowledge.hasSizes = true;
   if (matchPattern(op.pad(), m_TorchConstantIntList(padInts))){
     fill in sizes here....
   } else {
     knowledge.sizes.resize(inputRank, kUnknownSize);
   }
}
```